app/models/user.rb

```diff
@@ -212,4 +224,12 @@ def send_devise_notification(notification, *args)
   def ensure_roles
     self.roles = roles & ENUMS::USER_ROLES.values
   end
+
+  def ensure_referral_code
+    self.referral_code ||= begin
+      code = SecureRandom.urlsafe_base64(8)
```

Comment:
Is this line needed?

Reply:
Yes. It looks redundant but is required to set the initial value.